core/tests/widgets/test_slider.py

```diff
@@ -1,203 +1,469 @@
-        self.slider.focus()
-        self.assertActionPerformed(self.slider, "focus")
-
-    def test_set_on_press(self):
-        on_press = mock.Mock()
-        slider = toga.Slider(on_press=on_press)
-        self.assertEqual(slider.on_press._raw, on_press)
-
-    def test_set_on_release(self):
-        on_release = mock.Mock()
-        slider = toga.Slider(on_release=on_release)
-        self.assertEqual(slider.on_release._raw, on_release)
-
-    def assert_slider_value(self, tick_value, value, on_change_call_count=0):
-        self.assertEqual(self.slider.tick_value, tick_value)
-        self.assertEqual(self.slider.value, value)
-        self.assertValueSet(self.slider, "value", value)
-        self.assertEqual(
-            self.on_change.call_count,
-            on_change_call_count,
-            msg="on_changer call count is different than expected",
-        )
-
-    def assert_set_range(self, min_val, max_val):
-        self.slider.range = (min_val, max_val)
-        self.assertEqual(self.slider.min, min_val)
-        self.assertEqual(self.slider.max, max_val)
-        self.assertEqual(self.slider.range, (min_val, max_val))
-        self.assertValueSet(self.slider, "range", (min_val, max_val))
-
-    ######################################################################
-    # 2022-07: Backwards compatibility
-    ######################################################################
-
-    def test_init_with_deprecated(self):
-        # default is a deprecated argument
-        with self.assertWarns(DeprecationWarning):
-            my_slider = toga.Slider(
-                default=self.value,
-                range=self.range,
-                on_change=self.on_change,
-                enabled=self.enabled,
-                tick_count=self.tick_count,
-            )
-        self.assertEqual(my_slider.value, self.value)
-
-        # can't specify both default *and* value
-        with self.assertRaises(ValueError):
-            toga.Slider(
-                default=self.value,
-                value=self.value,
-                range=self.range,
-                on_change=self.on_change,
-                enabled=self.enabled,
-                tick_count=self.tick_count,
-            )
-
-    ######################################################################
-    # End backwards compatibility.
-    ######################################################################
+from toga_dummy.utils import assert_action_performed, attribute_value
+
+INITIAL_VALUE = 50
+INITIAL_MIN = 0
+INITIAL_MAX = 100
+INITIAL_TICK_VALUE = 6
+INITIAL_TICK_COUNT = 11
+INITIAL_ENABLED = True
+
+
+@fixture
+def on_change():
+    return Mock()
+
+
+@fixture
+def slider(on_change):
+    return toga.Slider(
+        value=INITIAL_VALUE,
+        range=(INITIAL_MIN, INITIAL_MAX),
+        on_change=on_change,
+        enabled=INITIAL_ENABLED,
+        tick_count=INITIAL_TICK_COUNT,
+    )
+
+
+def test_widget_created(slider, on_change):
```

Comment:
Test methods should have short descriptions of what they're asserting.

Reply:
Done, except for those methods whose names are self-explanatory.